app/models/event.rb

```diff
@@ -12,6 +12,8 @@
 class Event < ActiveRecord::Base
   translates :title, :introduction, :conclusion
   belongs_to :location
+  # suggestion from Nicholas
+  to_param :title
```

Comment:
@benichu are you trolling me? 😈 The class method does exactly what you described above.